maketeams3.py

```diff
@@ -76,177 +69,197 @@ def getPlayer(self, board):
+    for n, board in enumerate(players_split):
+        for player in board:
+            if player.alt:
+                alts_split[n].append(player)
+    for n, board in enumerate(alts_split):
+        for alt in board:
+            players_split[n].remove(alt)
+
+    # ensure the number of teams doesn't exceed the number of players available
+    # on a given board
+    num_teams = min([num_teams] + [len(board) for board in players_split])
+
+    # separate latest joining players into alternate lists as required
+    for n, board in enumerate(players_split):
+        board.sort(key=lambda player: player.date)
+        alts = board[num_teams:]
+        for p in alts:
+            alts_split[n].append(p)
+        del board[num_teams:]
+        board.sort(key=lambda player: player.rating, reverse=True)
+
+    players = sum(players_split,[])
+    #print len(players)
+    #print num_teams
+    #print alts_split
+
+    # snake draft players into initial teams and update player and team attributes
+    for board in players_split[1::2]:
+        board.reverse()
+    for n, board in enumerate(players_split):
+        for player in board:
+            player.board = n
+    teams = []
+    for n in range(num_teams):
+        teams.append(Team())
+    for n, board in enumerate(players_split):
+        for team, player in enumerate(board):
+            teams[team].changeBoard(n, player)
+
+    # convert players' friends and avoid from name to references of the friend's/avoid's player object
+    for player in players:
+        if player.friends or player.avoid:
+            player.friends = re.split("[^-_a-zA-Z0-9]+", player.friends) # separate friends requests into individual usernames - split on any number of non-(alphanumeric, hyphen or underscore)
+            player.avoid = re.split("[^-_a-zA-Z0-9]+", player.avoid) # separate avoid requests into individual usernames - split on any number of non-(alphanumeric, hyphen or underscore)
+        else:
+            player.friends = []
+            player.avoid = []
+    for player in players:
+        temp_friends = []
+        for friend in player.friends:
+            for potentialfriend in players:
+                if friend.lower() == potentialfriend.name.lower() and potentialfriend not in temp_friends: # prevent duplicated friend error
+                    temp_friends.append(potentialfriend)
+        player.friends = temp_friends
+
+        temp_avoid = []
+        for avoid in player.avoid:
+            for potentialavoid in players:
+                if avoid.lower() == potentialavoid.name.lower() and potentialavoid not in temp_avoid: # prevent duplicated friend error
+                    temp_avoid.append(potentialavoid)
+        player.avoid = temp_avoid
+
+    #remove friend requests for same board
+    for player in players:
+        for friend in player.friends:
+            if friend.board == player.board:
+                player.friends.remove(friend)
+    updatePref(players, teams)
```

Comment:
I don't see the benefit in passing as parameters instead of using the global. Also slightly confusing if it's the same name but they could potentially have different values?